src/libcrun/linux.c

```diff
@@ -1761,6 +1761,42 @@ libcrun_do_pivot_root (libcrun_container_t *container, bool no_pivot, const char
   return 0;
 }
 
+/* If one of stdin, stdout, stderr are pointing to /dev/null on
+ * the outside of the container, this moves it to /dev/null inside
+ * of the container. This needs to run afer pivot/chroot-ing. */
+int
+libcrun_reopen_dev_null(libcrun_error_t *err)
+{
+  struct stat dev_null;
+  struct stat statbuf;
+  int fd;
```

Comment:
you could declare fd as: `cleanup_close int fd = -1;`

In this way it will be automatically closed when it gets out of scope and you won't need to close it at all.